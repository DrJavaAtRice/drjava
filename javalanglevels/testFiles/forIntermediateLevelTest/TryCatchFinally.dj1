class TryCatchFinally {
  public int booyah() throws ClassNotFoundException{
    int i = 0,c = 6,j = 7;
    if (i+c < j) {
      throw new ClassNotFoundException("Someone had better catch me!");
    }
    try {
      if (c < j) {
        throw new CloneNotSupportedException("And someone had better catch me too!");
      }
      else throw new RuntimeException("No one needs to catch me");
    }
    catch (RuntimeException e) {
    }
    
    finally {
      return 5;
    }
  }
}