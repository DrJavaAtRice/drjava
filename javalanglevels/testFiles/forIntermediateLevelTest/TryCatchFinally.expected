// Language Level Converter line number map: dj*->java. Entries: 20
//     1->5         2->6         3->7         4->8         5->9         6->10        7->11        8->12   
//     9->13       10->14       11->15       12->16       13->17       14->18       15->19       16->20   
//    17->21       18->22       19->23       20->24   
class TryCatchFinally {
  public int booyah() throws ClassNotFoundException{
    final int i = 0,c = 6,j = 7;
    if (i+c < j) {
      throw new ClassNotFoundException("Someone had better catch me!");
    }
    try {
      if (c < j) {
        throw new CloneNotSupportedException("And someone had better catch me too!");
      }
      else throw new RuntimeException("No one needs to catch me");
    }
    catch (final RuntimeException e) {
    }
    
    finally {
      return 5;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public TryCatchFinally() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      TryCatchFinally cast = ((TryCatchFinally) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}
